src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -56,7 +196,35 @@ private function exit_element( $element ) {
 		parent::release_bookmark( "{$this->depth}_{$element}" );
 	}
 
-	private function opened_element() {
+	/**
+	 * @see https://html.spec.whatwg.org/#close-a-p-element
+	 * @return void
+	 */
+	private function close_p_element() {
+		$this->generate_implied_end_tags( 'P' );
+
+
+	}
+
+	/**
+	 * @TODO: Implement this
+	 *
+	 * @see https://html.spec.whatwg.org/#generate-implied-end-tags
+	 *
+	 * @param string|null $except_for_this_element Perform as if this element doesn't exist in the stack of open elements.
+	 * @return void
+	 */
+	private function generate_implied_end_tags( $except_for_this_element = null ) {
```

Comment:
Nice! I can't remember why I used an array here, but it doesn't seem like one is needed.

Reply:
you support "comprehensive" implied tags in the same function